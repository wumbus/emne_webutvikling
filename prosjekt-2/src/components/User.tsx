import React from "react";

export function title(value: number) {
    if (value == 40) {
        return "Maintainer"
    } else if (value == 50) {
        return "Owner"
    } else if (value == 30) {
        return "Developer"
    }
}

export function MembersList(props: { members: any, sort: any, filterBy: any }) {
    const membersOrg = props.members;
    const sorting = props.sort;
    const filterBy = props.filterBy;

    console.log(sorting);
    console.log(filterBy);
    console.log(membersOrg);
    console.log(membersOrg[0]);

    let members: any = [];
    if (membersOrg[0] != 1) { // Checks if members is set to the standard one or if there are actually members
        if (filterBy[0]) {
            membersOrg.forEach((member: any) => {
                if (member[3] == "30" && !members.includes(member)) {
                    members.push(member)
                }
            });
        }
        if (filterBy[1]) {
            membersOrg.forEach((member: any) => {
                if (member[3] == "40" && !members.includes(member)) {
                    members.push(member)
                }
            });
        }
        console.log(filterBy[2]);
    
        if (filterBy[2]) {
            membersOrg.forEach((member: any) => {
                if (member[3] == "50" && !members.includes(member)) {
                    members.push(member)
                }
            });
        }
        if (filterBy[3]) {
            // Bots
            membersOrg.forEach((member:any) => {
                console.log(member[0]);
                
                if (member[0].includes("_bot")  && !members.includes(member)) {
                    members.push(member)
                }
            });
        }
    
        // if none are selected show all
        if (!(filterBy[0] || filterBy[1] || filterBy[2] || filterBy[3])) {
            membersOrg.forEach((member: any) => {
                members.push(member)
            });
        }
    
    }
    
    console.log(members);


    if (sorting == "number") {
        console.log("fakk 2");
        members.sort((a: any, b: any) => b[3] - a[3]);
    } else if (sorting == "name") {
        members.sort();
    }



    return (
        <table>
            <th>Avatar</th>
            <th>Name</th>
            <th>Username</th>
            <th>Role</th>
            {members.map((member: any) =>
                <MemberItem key={member[0]} value={member} />)}
        </table>
    );
}

export function MemberItem(props: any) {
    return (<tr> <td><img src={props.value[2]} alt="Avatar" /> </td>
       <td>{props.value[1]} </td> <td>{props.value[0]} </td> <td>{title(props.value[3])}</td></tr>);
}

