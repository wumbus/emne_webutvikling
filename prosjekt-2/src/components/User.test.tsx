import React from "react";
import renderer from 'react-test-renderer';

import { MembersList, title } from './User';

test("Returns correct role", () => {
    expect(title(40)).toBe("Maintainer")
});


test("Snapshot for memberslist", () => {

    const members_given : any = [['Vahideh Rezaei', 'https://secure.gravatar.com/avatar/5a096b00ec3bbd22d9f3d761c3f23cfa?s=80&d=identicon', '40'],
    ['cldahl', 'Cecilie Le Duc Dahl', 'https://secure.gravatar.com/avatar/11b6eb38d8e618e9202a6f944526b065?s=80&d=identicon', '40'], 
    ['lisebul', 'Lise Birgitte Ullev√•lseter', 'https://secure.gravatar.com/avatar/6845d366a85cc4850f6771518755afbf?s=80&d=identicon', '40'],
    ['jimmynl', 'Jimmy Ngo Luong', 'https://secure.gravatar.com/avatar/b98569d860ff48aa6a8212857469a82e?s=80&d=identicon', '40'],
    ['project_17464_bot', 'Maintainer test token', 'https://secure.gravatar.com/avatar/a877b6393c1fa870c6d3c4cb9c6f34d8?s=80&d=identicon', '40'],
    ['group_28525_bot', 'Jonatan devver', 'https://secure.gravatar.com/avatar/e0b7100e71618df8efdd8d28f32410fc?s=80&d=identicon', '30']];


    let filterBy : any = [false, true, false, false];

    let tree : any = renderer.create(<MembersList members={members_given} sort="name" filterBy={filterBy}/>);

    expect(tree).toMatchSnapshot();
})

let filter: any = [true, false, false, false]



test("Snapshot for memberslist, developer", () => {

    const members: any = [['group_28525_bot', 'Jonatan devver', 'https://secure.gravatar.com/avatar/e0b7100e71618df8efdd8d28f32410fc?s=80&d=identicon', '30']];
    let filterBy : any = [true, false, false, false];

    let tree : any = renderer.create(<MembersList members={members} sort="name" filterBy={filterBy}/>);

    expect(tree).toMatchSnapshot();
})


test("Returns correct role", () => {
    expect(title(30)).toBe("Developer")
});








