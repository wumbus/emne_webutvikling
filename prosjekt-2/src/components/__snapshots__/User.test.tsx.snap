// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot for memberslist 1`] = `
<table>
  <th>
    Avatar
  </th>
  <th>
    Name
  </th>
  <th>
    Username
  </th>
  <th>
    Role
  </th>
  <tr>
     
    <td>
      <img
        alt="Avatar"
        src="https://secure.gravatar.com/avatar/11b6eb38d8e618e9202a6f944526b065?s=80&d=identicon"
      />
       
    </td>
    <td>
      Cecilie Le Duc Dahl
       
    </td>
     
    <td>
      cldahl
       
    </td>
     
    <td>
      Maintainer
    </td>
  </tr>
  <tr>
     
    <td>
      <img
        alt="Avatar"
        src="https://secure.gravatar.com/avatar/b98569d860ff48aa6a8212857469a82e?s=80&d=identicon"
      />
       
    </td>
    <td>
      Jimmy Ngo Luong
       
    </td>
     
    <td>
      jimmynl
       
    </td>
     
    <td>
      Maintainer
    </td>
  </tr>
  <tr>
     
    <td>
      <img
        alt="Avatar"
        src="https://secure.gravatar.com/avatar/6845d366a85cc4850f6771518755afbf?s=80&d=identicon"
      />
       
    </td>
    <td>
      Lise Birgitte Ullev√•lseter
       
    </td>
     
    <td>
      lisebul
       
    </td>
     
    <td>
      Maintainer
    </td>
  </tr>
  <tr>
     
    <td>
      <img
        alt="Avatar"
        src="https://secure.gravatar.com/avatar/a877b6393c1fa870c6d3c4cb9c6f34d8?s=80&d=identicon"
      />
       
    </td>
    <td>
      Maintainer test token
       
    </td>
     
    <td>
      project_17464_bot
       
    </td>
     
    <td>
      Maintainer
    </td>
  </tr>
</table>
`;

exports[`Snapshot for memberslist, developer 1`] = `
<table>
  <th>
    Avatar
  </th>
  <th>
    Name
  </th>
  <th>
    Username
  </th>
  <th>
    Role
  </th>
  <tr>
     
    <td>
      <img
        alt="Avatar"
        src="https://secure.gravatar.com/avatar/e0b7100e71618df8efdd8d28f32410fc?s=80&d=identicon"
      />
       
    </td>
    <td>
      Annen devver
       
    </td>
     
    <td>
      group_28625_bot
       
    </td>
     
    <td>
      Developer
    </td>
  </tr>
  <tr>
     
    <td>
      <img
        alt="Avatar"
        src="https://secure.gravatar.com/avatar/e0b7100e71618df8efdd8d28f32410fc?s=80&d=identicon"
      />
       
    </td>
    <td>
      Jonatan devver
       
    </td>
     
    <td>
      group_28525_bot
       
    </td>
     
    <td>
      Developer
    </td>
  </tr>
</table>
`;
